src/components/common/TanStackTable/MadieTable.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useState } from "react";
+import DataTypeCell from "./DataTypeCell";
+import { DataElement } from "cqm-models";
+
+import "./MadieTable.scss";
+
+import {
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+
+import useQdmExecutionContext from "../../routes/qdm/useQdmExecutionContext";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import TimingCell from "./TimingCell";
+
+const columnHelper = createColumnHelper<DataElement>();
+
+interface MadieTableProps {
+  dataElements?: DataElement[];
+  onView?: (dataElement: DataElement) => void;
+}
+const MadieTable = ({ dataElements = [], onView }: MadieTableProps) => {
```

Comment:
MadieTable seems more general. Can we name it something else probably DataElementTable? other DataelementTable component is just the container for this table. if it makes sense we could merge these two.